### PR TITLE
const and ubyte added as described in documentation to libpq

### PR DESCRIPTION
**Pull request by denizzzka**

### CODE REVIEW

**Comment by aldacron** [comment]
First, which version of the documentation are you following? The master branch is 9.3.

Second, please squash all the commits. There's six commits here for something that should really only be one.

If the versions match and the commits are squashed, I'll merge.

**Comment by denizzzka** [comment]
> First, which version of the documentation are you following? The master branch is 9.3.

Latest stable PostgreSQL is 9.5.2

> Second, please squash all the commits. There's six commits here for something that should really only be one.

Well, a little later

**Comment by aldacron** [comment]
> Latest stable PostgreSQL is 9.5.2

Then please verify that there are no differences with the docs for 9.3. I don't expect that there would be, but I'd just like to be on the safe side. At some point I'll get the bindings up to date.

File: source/derelict/pq/pq.d [hunk 1]
```diff
@@ -257,7 +257,7 @@ extern(C) @nogc nothrow {
     alias da_PQconnectionNeedsPassword = int function(const(PGconn)*);
     alias da_PQconnectionUsedPassword = int function(const(PGconn)*);
     alias da_PQclientEncoding = int function(const(PGconn)*);
-    alias da_PQsetClientEncoding = int function(PGconn*,char*);
+    alias da_PQsetClientEncoding = int function(PGconn*,const(char)*);
```
Comment (denizzzka) [inline source/derelict/pq/pq.d]: same as in 9.3

File: source/derelict/pq/pq.d [hunk 2]
```diff
@@ -273,13 +273,13 @@ extern(C) @nogc nothrow {
     alias da_PQregisterThreadLock = pgthreadlock_t function(pgthreadlock_t);
 
     alias da_PQexec = PGresult* function(PGconn*,const(char)*);
-    alias da_PQexecParams = PGresult* function(PGconn*,const(char)*,int,Oid*,const(char*)*,int*,int*,int);
-    alias da_PQprepare = PGresult* function(PGconn*,char*,char*,int,Oid*);
+    alias da_PQexecParams = PGresult* function(PGconn*,const(char)*,int,const Oid*,const(ubyte*)*,const int*,const int*,int);
```
Comment (denizzzka) [inline source/derelict/pq/pq.d]: same as in 9.3

File: source/derelict/pq/pq.d [hunk 3]
```diff
@@ -273,13 +273,13 @@ extern(C) @nogc nothrow {
...
+    alias da_PQprepare = PGresult* function(PGconn*,const(char)*,const(char)*,int,Oid*);
```
Comment (denizzzka) [inline source/derelict/pq/pq.d]: should be const Oid *paramTypes

File: source/derelict/pq/pq.d [hunk 4]
```diff
@@ -273,13 +273,13 @@ extern(C) @nogc nothrow {
...
     alias da_PQexecPrepared = PGresult* function(PGconn*,const(char)*,int,const(char*)*,const(int)*,const(int)*,int);
     alias da_PQsendQuery = int function(PGconn*,const(char)*);
-    alias da_PQsendQueryParams = int function(PGconn*,const(char)*,int,Oid*,const(char*)*,int*,int*,int);
+    alias da_PQsendQueryParams = int function(PGconn*,const(char)*,int,const Oid*,const(ubyte*)*,const int*,const int*,int);
```
Comment (denizzzka) [inline source/derelict/pq/pq.d]: same as in 9.3.

File: source/derelict/pq/pq.d [hunk 5]
```diff
@@ -273,13 +273,13 @@ extern(C) @nogc nothrow {
...
     alias da_PQsendPrepare = int function(PGconn*,const(char)*,const(char)*,int,Oid*);
-    alias da_PQsendQueryPrepared = int function(PGconn*,const(char)*,int,const(char*)*,int*,int*,int);
+    alias da_PQsendQueryPrepared = int function(PGconn*,const(char)*,int,const(ubyte*)*,const(int)*,const(int)*,int);
```
Comment (denizzzka) [inline source/derelict/pq/pq.d]: same as in 9.3

File: source/derelict/pq/pq.d [hunk 6]
```diff
@@ -332,8 +332,8 @@ extern(C) @nogc nothrow {
     alias da_PQnparams = int function(const(PGresult)*);
     alias da_PQparamtype = Oid function(const(PGresult)*,int);
 
-    alias da_PQdescribePrepared = PGresult* function(PGconn*,char*);
-    alias da_PQdescribePortal = PGresult* function(PGconn*,char*);
+    alias da_PQdescribePrepared = PGresult* function(PGconn*,const(char)*);
```
Comment (denizzzka) [inline source/derelict/pq/pq.d]: same as in 9.3

File: source/derelict/pq/pq.d [hunk 7]
```diff
@@ -332,8 +332,8 @@ extern(C) @nogc nothrow {
...
+    alias da_PQdescribePortal = PGresult* function(PGconn*,const(char)*);
```
Comment (denizzzka) [inline source/derelict/pq/pq.d]: same as in 9.3

File: source/derelict/pq/pq.d [hunk 8]
```diff
@@ -273,13 +273,13 @@ extern(C) @nogc nothrow {
     alias da_PQregisterThreadLock = pgthreadlock_t function(pgthreadlock_t);
 
     alias da_PQexec = PGresult* function(PGconn*,const(char)*);
-    alias da_PQexecParams = PGresult* function(PGconn*,const(char)*,int,Oid*,const(char*)*,int*,int*,int);
-    alias da_PQprepare = PGresult* function(PGconn*,char*,char*,int,Oid*);
+    alias da_PQexecParams = PGresult* function(PGconn*,const(char)*,int,const Oid*,const(ubyte*)*,const int*,const int*,int);
+    alias da_PQprepare = PGresult* function(PGconn*,const(char)*,const(char)*,int,const(Oid)*);
```
Comment (denizzzka) [inline source/derelict/pq/pq.d]: same as in 9.3

File: source/derelict/pq/pq.d [hunk 9]
```diff
@@ -273,13 +273,13 @@ extern(C) @nogc nothrow {
...
     alias da_PQexecPrepared = PGresult* function(PGconn*,const(char)*,int,const(char*)*,const(int)*,const(int)*,int);
     alias da_PQsendQuery = int function(PGconn*,const(char)*);
-    alias da_PQsendQueryParams = int function(PGconn*,const(char)*,int,Oid*,const(char*)*,int*,int*,int);
+    alias da_PQsendQueryParams = int function(PGconn*,const(char)*,int,const Oid*,const(ubyte*)*,const int*,const int*,int);
```
Comment (denizzzka) [inline source/derelict/pq/pq.d]: same as in 9.3

File: source/derelict/pq/pq.d (hunk 5 above)
Comment (denizzzka) [inline source/derelict/pq/pq.d]: PQsendQueryPrepared

**Comment by denizzzka** [comment]
> Because DerelictPQ currently binds to 9.3

Why?

However, I checked - all coincided with 9.3 version

**Comment by denizzzka** [comment]
New release version?

**Comment by aldacron** [comment]
Yes, of course. I had to do some house keeping first. The branches in this repository were disorganized.

**Pull request merged**